test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1588 @@
+		]}}`
+		expected := []TestContainerInfo{
+			{
+				Name:      "c1",
+				Resources: &ContainerResources{CPUReq: "400m", CPULim: "400m", MemReq: "400Mi", MemLim: "400Mi"},
+			},
+		}
+		patchStringExceedCPU := `{"spec":{"containers":[
+			{"name":"c1", "resources":{"requests":{"cpu":"600m","memory":"200Mi"},"limits":{"cpu":"600m","memory":"200Mi"}}}
+		]}}`
+		patchStringExceedMemory := `{"spec":{"containers":[
+			{"name":"c1", "resources":{"requests":{"cpu":"250m","memory":"750Mi"},"limits":{"cpu":"250m","memory":"750Mi"}}}
+		]}}`
+
+		ginkgo.By("Creating a ResourceQuota")
+		_, rqErr := f.ClientSet.CoreV1().ResourceQuotas(f.Namespace.Name).Create(context.TODO(), &resourceQuota, metav1.CreateOptions{})
+		framework.ExpectNoError(rqErr, "failed to create resource quota")
+
+		isReadyToUse := false
+		waitErr := wait.PollUntilContextTimeout(ctx, 1*time.Second, 1*time.Minute, true, func(ctx context.Context) (done bool, err error) {
+			quota, err := f.ClientSet.CoreV1().ResourceQuotas(f.Namespace.Name).Get(context.TODO(), rcname, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			podQuota := quota.Status.Hard[v1.ResourcePods]
+			quantity := resource.MustParse("2")
+			isReadyToUse = (&podQuota).Cmp(quantity) == 0
+			return isReadyToUse, nil
+		})
+
+		if !isReadyToUse {
+			msg := fmt.Sprintf("resource quota " + rcname + " never synced,")
+			if waitErr != nil {
+				msg += fmt.Sprintf(":message: %v", waitErr.Error())
+			}
+			err := fmt.Errorf(msg)
+			framework.ExpectNoError(err)
+		}
+
+		tStamp := strconv.Itoa(time.Now().Nanosecond())
+		initDefaultResizePolicy(containers)
+		initDefaultResizePolicy(expected)
+		testPod1 := makeTestPod(f.Namespace.Name, "testpod1", tStamp, containers)
+		testPod2 := makeTestPod(f.Namespace.Name, "testpod2", tStamp, containers)
+
+		ginkgo.By("creating pods")
+		newPod1 := podClient.CreateSync(ctx, testPod1)
+		newPod2 := podClient.CreateSync(ctx, testPod2)
+
+		ginkgo.By("verifying the pod is in kubernetes")
+		selector := labels.SelectorFromSet(labels.Set(map[string]string{"time": tStamp}))
+		options := metav1.ListOptions{LabelSelector: selector.String()}
+		podList, listErr := podClient.List(context.TODO(), options)
+		framework.ExpectNoError(listErr, "failed to query for pods")
+		gomega.Expect(podList.Items).Should(gomega.HaveLen(2))
+
+		ginkgo.By("verifying initial pod resources, allocations, and policy are as expected")
+		verifyPodResources(newPod1, containers)
+
+		ginkgo.By("patching pod for resize within resource quota")
+		patchedPod, pErr := f.ClientSet.CoreV1().Pods(newPod1.Namespace).Patch(context.TODO(), newPod1.Name,
+			types.StrategicMergePatchType, []byte(patchString), metav1.PatchOptions{})
+		framework.ExpectNoError(pErr, "failed to patch pod for resize")
+
+		ginkgo.By("verifying pod patched for resize within resource quota")
+		verifyPodResources(patchedPod, expected)
+		verifyPodAllocations(patchedPod, containers, true)
```

Comment:
async?

Reply:
Thanks, acknowledged will try it and commit your suggestion if it passes.

Reply:
Done, with use of Eventually. Please let me know if this sufficient.